docs/content/sips/017-configuring-and-emitting-observability.md

```diff
@@ -0,0 +1,85 @@
+
+Created: February 27, 2024
+
+Updated: February 27, 2024
+
+## Background
+
+[Observability](https://opentelemetry.io/docs/concepts/observability-primer/#what-is-observability) is critical for a great developer experience. Improving the observability of Spin can be broken down into four categories:
+
+1. Runtime observability: Observing the Spin runtime itself e.g. spans and metrics around important parts of the Spin runtime.
+2. Trigger observability: Observing the requests made to Spin applications e.g. spans and metrics around the requests made to Spin applications.
+3. Component observability: Observing the interaction between composed components e.g having Wasmtime auto-instrument Wasm component graphs to produce spans.
+4. Guest observability: Observing code within the guest module e.g. building a host component for WASI Observe.
+
+More detail on these categories can be found [here](https://github.com/fermyon/spin/issues/2293).
+
+## Proposal
+
+This SIP aims to improve the runtime and trigger observability of Spin. First, it will suggest how observability in Spin can be configured. Second, it will present the specific runtime and trigger observability we want to emit. Finally, it will outline a plan going forward for how observability should be added to new Spin features.
+
+This proposal assumes that all observability data produced should be OTEL compliant.
+
+### Configuring observability
+
+The developer must be able to configure the endpoints of OTLP compliant collectors where traces and metrics can be sent. This can be expressed in the `spin.toml` manifest.
+
+```toml
+[application]
+otlp_grpc_traces_endpoint = "http://localhost:4317"
+otlp_grpc_metrics_endpoint = "http://localhost:4317"
+```
+
+**INPUT NEEDED:** How do we want to configure these endpoints? Where do they belong in the manifest? Do we want to allow for separate endpoints for traces and metrics? Do we want to allow for choosing between gRPC and HTTP? Should we let them pass a flag to `spin` to override the manifest value?
+
+The developer must be able to opt in or out of trace context propagation. This should be possible on a per component basis. By default trace context propagation is disabled for a component. This can be expressed in the `spin.toml` manifest.
+
+```toml
+[component.my-component]
+allow_trace_context_propagation = true
+```
+
+**INPUT NEEDED:** Is it correct to put this on the component config as opposed to the trigger config?
+
+Improving the guest observability of Spin with WASI Observe is outside the scope of this SIP. But, it is worth noting that when we add guest observability we may want to move from a simple binary model of trace context propagation to an allow-list based system.
+
+```toml
+[component.my-component]
+allowed_trace_context_propagation_origins = ["my-component", "my-other-component"]
+```
+
+**INPUT NEEDED:** I know this is something we need but I don't actually have any clue what it would look like.
```

Comment:
Fine to punt on this for now I think, though we should clearly document that propagation is all-or-nothing.